### PR TITLE
Working dmd on OpenBSD.

### PR DESCRIPTION
**Pull request by ibara**
As mentioned on the forum/mailing list, it was suggested that I send these PRs.
I appreciate the Dlang team's willingness to accept OpenBSD patches.

These patches are enough to finish the dmd compiler on OpenBSD. It works and is able to rebuild a working copy of itself.

I have only tested the openbsd-64 version. 32-bit I will eventually be able to test, but it will likely be a few months.

### CODE REVIEW

File: src/dmd/globals.h [hunk 1]
```diff
@@ -34,8 +34,8 @@ enum
     TargetOS_DragonFlyBSD = 0x40,
 
     // Combination masks
-    all = TargetOS_linux | TargetOS_Windows | TargetOS_OSX | TargetOS_FreeBSD | TargetOS_Solaris | TargetOS_DragonFlyBSD,
-    Posix = TargetOS_linux | TargetOS_OSX | TargetOS_FreeBSD | TargetOS_Solaris | TargetOS_DragonFlyBSD,
```
Comment (thewilsonator) [inline src/dmd/globals.h:38]: you need to also update `frontend.h` with these changes

File: src/dmd/backend/elfobj.d [hunk 2]
```diff
@@ -3378,10 +3392,26 @@ private void obj_rtinit()
     // make the symbols hidden so that each DSO gets its own brackets
     IDXSYM minfo_beg, minfo_end, dso_rec;
 
+    version (OpenBSD)
```
Comment (thewilsonator) [inline src/dmd/backend/elfobj.d:3395]: is there any problem in unconditionally emitting `deh_beg` and `deh_end` (and thus remove all the `version (OpenBSD)`?

File: src/dmd/globals.h (hunk 1 above)
Comment (ibara) [reply src/dmd/globals.h:38]: Ah. Didn't notice that. Will update shortly.

File: src/dmd/backend/elfobj.d (hunk 2 above)
Comment (ibara) [reply src/dmd/backend/elfobj.d:3395]: I'm not actually sure about this. This is the place I was hoping to rely on the expertise of the community. My original version of the diff did unconditionally do this. But as I don't have other platforms to test with, I erred on the side of being overly cautious to not alter any other platform.

Comment (ibuclaw) [reply src/dmd/backend/elfobj.d:3395]: Use `static if (TARGET_OPENBSD)`

**Review by ibara — COMMENTED** [review]
I don't know why GitHub is not letting me reply directly, but I did what @ibuclaw suggested in `src/dmd/backend/elfobj.d`.

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**